src/integration/Privatisation.spec.ts

```diff
@@ -0,0 +1,514 @@
+  homepagePageService,
+  sitesService,
+  deploymentsService,
+  gitHubService,
+})
+const SettingsRouter = new _SettingsRouter({ settingsService })
+const settingsSubrouter = SettingsRouter.getRouter()
+const subrouter = express()
+subrouter.use("/:siteName", settingsSubrouter)
+
+const mockGenericAxios = mockAxios.create()
+mockGenericAxios.patch.mockResolvedValue({
+  data: [],
+})
+// const migrateSpy = jest
+//   .spyOn(ReviewsRouter, "checkIfSiteIsUnmigrated")
+//   .mockResolvedValue(true)
+
+const FRONTEND_URL = process.env.FRONTEND_URL || "http://localhost:3000"
+
+describe("Password integration tests", () => {
+  beforeAll(async () => {
+    // NOTE: Because SitesService uses an axios instance,
+    // we need to mock the axios instance using es5 named exports
+    // to ensure that the calls for .get() on the instance
+    // will actually return a value and not fail.
+    jest.mock("../services/api/AxiosInstance.ts", () => ({
+      __esModule: true, // this property makes it work
+      genericGitHubAxiosInstance: mockGenericAxios,
+    }))
+
+    // We need to force the relevant tables to start from a clean slate
+    // Otherwise, some tests may fail due to the auto-incrementing IDs
+    // not starting from 1
+    await User.sync({ force: true })
+    await Site.sync({ force: true })
+    await Repo.sync({ force: true })
+    await SiteMember.sync({ force: true })
+    await Deployment.sync({ force: true })
+
+    await User.create(MOCK_USER_DBENTRY_ONE)
+    await User.create(MOCK_USER_DBENTRY_TWO)
+    await User.create(MOCK_USER_DBENTRY_THREE)
+    await Site.create(MOCK_SITE_DBENTRY_ONE)
+    await Site.create(MOCK_SITE_DBENTRY_TWO)
+    await Deployment.create(MOCK_DEPLOYMENT_DBENTRY_ONE)
+    await Deployment.create(MOCK_DEPLOYMENT_DBENTRY_TWO)
+    await Repo.create(MOCK_REPO_DBENTRY_ONE)
+    await Repo.create(MOCK_REPO_DBENTRY_TWO)
+    await SiteMember.create(MOCK_SITEMEMBER_DBENTRY_ONE)
+    await SiteMember.create(MOCK_SITEMEMBER_DBENTRY_TWO)
+  })
+
+  afterAll(async () => {
+    await User.sync({ force: true })
+    await Site.sync({ force: true })
+    await Repo.sync({ force: true })
+    await SiteMember.sync({ force: true })
+    await Deployment.sync({ force: true })
+  })
+
+  describe("GET /repo-password", () => {
+    it("should successfully retrieve password for an amplify site", async () => {
+      // Arrange
+      const app = generateRouterForUserWithSite(
+        subrouter,
+        MOCK_USER_SESSION_DATA_TWO,
+        MOCK_REPO_NAME_TWO
+      )
+
+      const expected = {
+        password: MOCK_PASSWORD,
+        isAmplifySite: true,
+      }
+
+      // Act
+      const actual = await request(app).get(
+        `/${MOCK_REPO_NAME_TWO}/repo-password`
+      )
+
+      // Assert
+      expect(actual.statusCode).toEqual(200)
+      expect(actual.body).toMatchObject(expected)
+    })
+
+    it("should successfully return empty string for an amplify site with no password", async () => {
```

Comment:
can we have another test case where we want to remove but we also provide a password?

Reply:
We actually already have this - it's the above test for invalid IV, I've updated the name of the test to make it clearer! https://github.com/isomerpages/isomercms-backend/pull/807/commits/79f7fce7969a0c816a0893ee2babad6b253e5fa3